pxr/base/tf/fileUtils.cpp

```diff
@@ -94,6 +93,21 @@ Tf_HasAttribute(
         }
         return false;
     }
+
+    // Ignore reparse points on network volumes. They can't be resolved
+    // properly, so simply remove the reparse point attribute and treat
+    // if like a regular file/directory.
+    if ((attribs & FILE_ATTRIBUTE_REPARSE_POINT) != 0) {
```

Comment:
This seems to make the assumption that a child of a reparse point still has the `FILE_ATTRIBUTE_REPARSE_POINT` attr set, but in my testing, it doesn't seem to:

https://gist.github.com/pmolodo/a48c1bbc96d7e2723b5ae2ffcdf29c1d

That is, if we have made a junction:

```
C:\path\to\some => D:\another\location
```
...and at the "real location", there is a subdirectory `D:\another\location\directory`, then

```
C:\path\to\some = has ReparsePoint attr
C:\path\to\some\directory = DOESN'T have ReparsePoint attr
```

It's possible that the behavior is different with network paths, though... or my assumption that the PowerShell attributes returned correspond exactly to the c function ones is wrong.  Or that I'm testing with "Junction" reparse points, and you were testing with "Symlink" reparse points?

Reply:
What you say is correct, but I don't see what it have to do with this PR? Here is how I (perhaps incorrectly) see things:

For clarity, I'm going to change your example so that "S:\path\to\some" -> "D:\another\location" (to remind us that "S" must be a network drive for this PR to have any effect, and recall that "D" is drive "D" on the remote machine and therefore inaccessible to the machine that has mounted drive "S"). So, before this PR, `S:\path\to\some\directory` would report as "not a reparse point", and after this PR `S:\path\to\some\directory` would report as "not a reparse point". Which is what I want.

This function is not where we do the recursive searching up the path for reparse points (nor does TfReadLink march up the path). That kind of marching up the path happens somewhere else. `resolveSymlinks` set to true passed into this function never has (and continue to not) resolve symlinks of ancestors of the specifically requested directory. So to summarize, the old behavior (with resolveSymLinks=true) was:
```
S:\path\to\some -> attempt to return attribs of D:\another\location (and fail, and therefore return "file doesn't exist")
S:\path\to\some\directory -> return attribs of S:\path\to\some\directory, which reports "not a reparse point"
```
and the new behavior is:
```
S:\path\to\some -> return attribs of S:\path\to\some (with reparse point attrib stripped off)
S:\path\to\some\directory -> unchanged (return attribs of S:\path\to\some\directory, which reports "not a reparse point")
```
At least this was the intent of this PR, and I _think_ it is in fact doing this...